kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResourceList.java

```diff
@@ -44,6 +45,19 @@
   @JsonProperty("metadata")
   private ListMeta metadata;
 
+  public CustomResourceList() {
+    try {
+      Class<T> customResourceClass = getCustomResourceClass();
+      if (customResourceClass != null) {
+        HasMetadata instance = customResourceClass.getDeclaredConstructor().newInstance();
```

Comment:
You don't need to instantiate the class here. You can just call `HasMetadata.getApiVersion(customResourceClass)` and `HasMetadata.getKind(customResourceClass)` instead.

Reply:
@metacosm I was just trying to be consistent with how it's down in CustomResource - there it is calling the instance method, which defaults to the static method.  Is there a chance that someone can override getKind?

Reply:
It's indeed possible… not likely, but possible. Good point.

Reply:
The thing is I'd like to avoid instantiating the CR if possible. Did you run into specific issues with the current implementation? If yes, how did you address them? Would using the first item work? Or do you need that information before the list is "filled"?

Reply:
> Did you run into specific issues with the current implementation? 

Not really, the empty api and version was noticed on a list returned by some of mock logic - it's not something that's needed by our implementation code.

> Would using the first item work?

My assumption was there are instances where the list may be empty.

Reply:
Empty list is indeed the issue. Then again, if there isn't an actual issue, I'm a little bit reluctant to add reflection-based instantiation because I worry of side-effects (in particular, for native compilation).

Reply:
I was hoping it might just be a quick thing to make CustomResourceList more consistent with CustomResource.  However it does require the developer to know to provide this override to supply the class and it's not something that we actually require at this point - so if there's reasons to not pursue this, I'm fine with that.